hicommonwealth/commonwealth
# Added newer `<CWCoverImageUploader/>` component with feature to switch b/w processed images

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/9606

## Description of Changes
- Added newer `<CWCoverImageUploader />` component with better form validations states and simpler logic.
- Removed older `<CWCoverImageUploader />` component.
- Added newer `<CWCoverImageUploader />` to component showcase
- Added a new feature to switch b/w multiple generate `canSwitchBetweenProcessedImages` prop is `true`. This will also correctly update form state based on the active image, if the control is hooked to form.

## "How We Fixed It"
N/A

## Test Plan
Note: All the variants are also added to the components page. Testing can be done from there as well.

- Visit any form that uses an image upload option that works correctly. ex: create community, edit profile, manage community, manage contest, launch token, etc
- Verify there are no regressions with the cover image uploader (Note: I intentionally removed some micro UIs that were redundant)
- Verify that when image behavior is cover, you can still see the generate image button. (see components page for this variant)
- Verify you are able to switch b/w multiple generated images when `canSwitchBetweenProcessedImages` prop is `true` (see components page for this variant)

## Deployment Plan
N/A

## Other Considerations
N/A
